src/main/java/com/qaprosoft/carina/demo/gui/pages/PhoneModelPage.java

```diff
@@ -0,0 +1,111 @@
+
+public class PhoneModelPage extends AbstractUIObject {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    @FindBy(xpath = "//div[@class='main main-review right l-box col']")
+    private ExtendedWebElement phoneModelBody;
+
+    @FindBy(xpath = "//div[@class='review-header']//ul/li/a[text()='Opinions']")
+    private ExtendedWebElement opinionsButton;
+
+    @FindBy(xpath = "//div[@class='sub-header centered']//select[@name='nSortNew']")
+    private ExtendedWebElement opinionsSort;
+
+    @FindBy(xpath = "//div[@id='all-opinions']/div[1]//a[@class='voting-link vote-up']")
+    private ExtendedWebElement likeFirstComment;
+
+    @FindBy(xpath = "//div[@id='all-opinions']/div[1]//a[@class='voting-link vote-down']")
+    private ExtendedWebElement dislikeFirstComment;
+
+    @FindBy(xpath = "//div[@id='all-opinions")
+    private List<ExtendedWebElement> comments;
+
+    @FindBy(xpath = "//div[@id='all-opinions']/div//span[@class='thumbs-score']")
+    private List<ExtendedWebElement> likesOnComments;
+
+    @FindBy(xpath = "//div[@id=\"all-opinions\"]//li/time")
+    private List<ExtendedWebElement> dateComments;
+
+    public PhoneModelPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public PhoneModelPage(WebDriver driver, SearchContext searchContext) {
+        super(driver, searchContext);
+    }
+
+    public boolean isPhoneModelPagePresented(){
+        return phoneModelBody.isPresent();
+    }
+
+    public void clickOpinions() {
+        opinionsButton.click();
+    }
+
+    public void sortByBestRating() {
+        opinionsSort.select("Best rating");
+    }
+
+    public boolean areCommentsSortedByRating() {
+        List<Integer> likes = new ArrayList<>();
+        likesOnComments.stream().forEach(t -> likes.add(Integer.parseInt(t.getText())));
+        List<Integer> likesBefore = likes;
+        likes.stream().sorted(Comparator.reverseOrder());
+        return likes == likesBefore;
+    }
+
+    public boolean isLikeFirstComment() {
+        int before = Integer.parseInt(likesOnComments.get(0).getText());
+        likeFirstComment.click();
+        int after = Integer.parseInt(likesOnComments.get(0).getText());
+        return (after > before);
+    }
+
+    public boolean isUnratedFirstComment() {
+        int before = Integer.parseInt(likesOnComments.get(0).getText());
+        dislikeFirstComment.click();
+        int after = Integer.parseInt(likesOnComments.get(0).getText());
+        return (before > after);
+    }
+
+    public void sortByNewFirst() {
+        opinionsSort.select("Newest first");
+    }
+
+    public boolean areCommentsSortedByDate() {
+        List<Date> dates = new ArrayList<>();
+        List<Integer> hours = new ArrayList<>();
+        dateComments.stream().forEach(t -> {
+            if (t.getText().contains("ago")) {
+                hours.add(Integer.parseInt(t.getText().replaceAll("\\D+", "")));
+            } else {
```

Comment:
you forgot minutes, separate this
add logic for 'minutes' and 'hours' separately